lib/system/web/src/keyboard_engine/mod.rs

```diff
@@ -0,0 +1,128 @@
+pub struct KeyboardEngine {
+    pub bindings: Rc<Bindings>,
+    pub state:    Rc<Cell<U256>>,
+    pub target:   web_sys::EventTarget
+}
+
+impl KeyboardEngine {
+    pub fn new(target: &web_sys::EventTarget) -> Self {
+        let mut this = Self {
+            bindings: Rc::new(Bindings::new()),
+            state:    Rc::new(Cell::new(U256::from(0))),
+            target:   target.clone()
+        };
+        this.init();
+        this
+    }
+
+    pub fn from_tag_name(html_tag: &str) -> Result<Self> {
+        let element = document()?
+            .get_elements_by_tag_name(html_tag)
+            .item(0).ok_or_else(|| Error::missing("there is no such element"));
+
+        Ok(KeyboardEngine::new(&element.unwrap()))
+    }
+
+    fn init(&mut self) {
+        self.bind_keydown_event();
+        self.bind_keyup_event();
+    }
+
+    fn bind_keydown_event(&self) {
+        let state = Rc::clone(&self.state);
+        let bindings = Rc::clone(&self.bindings);
+        let callback = Box::new(move |event: web_sys::KeyboardEvent| {
+            if event.repeat() {
+                return;
+            }
+
+            let key_code = event.key_code();
+            let bit = U256::from(1) << (key_code as usize);
+            let key = state.get() | bit;
+            state.set(key);
+            bindings.call_by_key(key);
+        });
+        let callback = Closure::wrap(callback as Box<dyn FnMut(_)>);
+
+        let callback_ref = callback.as_ref().unchecked_ref();
+        self.target
+            .add_event_listener_with_callback("keydown", callback_ref).unwrap();
+        callback.forget();
+    }
+
+    fn bind_keyup_event(&self) {
+        let bindings = Rc::clone(&self.bindings);
+        let state =  Rc::clone(&self.state);
+        let callback = Box::new(move |event: KeyboardEvent| {
+            let key_code = event.key_code();
+            let bit = U256::from(1) << (key_code as usize);
+            let key = state.get() ^ bit;
+            state.set(key);
+            bindings.call_by_key(key);
+        });
+        let callback = Closure::wrap(callback as Box<dyn FnMut(_)>);
+
+        let callback_ref = callback.as_ref().unchecked_ref();
+        self.target
+            .add_event_listener_with_callback("keyup", callback_ref).unwrap();
+        callback.forget();
+    }
+
+    /// Returns U256 number that represents bitstring of pressed keys
+    fn get_key_bits(&self, combination: Vec<u32>) -> U256 {
+        combination.iter()
+            .fold(U256::from(0), |acc, x| {
+                let bit = U256::from(1) << (*x as usize);
+                acc | bit
+            })
+    }
+
+    /// Capture key combination
+    ///
+    /// Uses u32 as key code from WebIDL
+    /// https://github.com/rustwasm/wasm-bindgen/blob/913fdbc3daff65952b5678a34b98e07d4e6e4fbb/crates/web-sys/webidls/enabled/KeyEvent.webidl
+    pub fn capture(
+        &self,
+        combination: Vec<u32>,
+        callback:    Box<dyn FnMut()>
```

Comment:
As I've written above, the `Box` is implementaion detail ,so the type here should be something like `F: FnMut()` or `F: Fn()` and should be boxed inside of `capture`. This way the interface is much more clean :)